client.go

```diff
@@ -1207,46 +1207,47 @@ func (t *Torrent) MergeSpec(spec *TorrentSpec) error {
 }
 
 func useTorrentSources(sources []string, t *Torrent) {
-	for _, s := range sources {
-		go func(s string) {
-			err := useTorrentSource(s, t)
-			if err != nil {
+	ctx := context.Background()
```

Comment:
It makes sense to make context part of useTorrentSource's signature, except we don't actually make use of it. Why not bind it to the lifetime of the Torrent or something, so it is cancelled if the torrent closes?

Reply:
Yeah that's definitely true; I just thought of it more as a style thing for future possible use cases of `useTorrentSource` (no idea what else it could be used for though). But, yeah, if you're sure that `useTorrentSource` will only ever be used by `useTorrentSources` then it would be best to remove the parameter altogether. It would be even better if we could restrict its visibility to only `useTorrentSources` actually.

The restricting visibility part would require moving `useTorrentSource` to a closure (since nested functions cant be done in Go), and definitely make things uglier. If you're OK with that, I'm all for it.

Reply:
No I'm fine with this, maybe a TODO to wire it in later to the lifetime of Torrent.